slsa_for_models/README.md

```diff
@@ -0,0 +1,85 @@
+provenance for all new models before serving them. The verification can also be
+done manually, on demand.
+
+As an additional benefit, having provenance for a model allows users to react
+to vulnerabilities in a training framework: they can quickly determine if a
+model needs to be retrained because it was created using a vulnerable version.
+
+## Usage
+
+We support both TensorFlow and PyTorch models. The example repo trains a model
+on [CIFAR10][cifar10] dataset, saves it in one of the supported formats, and
+generates provenance for the output. All of this happens during a [GitHub Actions
+workflow][workflow] which takes as input the format to save the model into. The
+supported formats are:
+
+| Workflow Argument            | Training Framework | Model format                    |
+|------------------------------|--------------------|---------------------------------|
+| `tensorflow_model.keras`     | TensorFlow         | Keras format (default)          |
+| `tensorflow_hdf5_model.h5`   | TensorFlow         | Legacy HDF5 format              |
+| `tensorflow_hdf5.weights.h5` | TensorFlow         | Legacy HDF5 weights only format |
+| `pytorch_model.pth`          | PyTorch            | PyTorch default format          |
+| `pytorch_full_model.pth`     | PyTorch            | PyTorch complete model format   |
+| `pytorch_jitted_model.pt`    | PyTorch            | PyTorch TorchScript format      |
+
+To test, fork this repository, then head over to the Actions tab and select the
+"SLSA for ML models example" workflow. Since the workflow has a
+`workflow_dispatch` trigger, it can be invoked on demand: click the `Run
+workflow` button, then select the value for the "Name of the model" argument.
+
+TODO: Screenshot of all the steps listed above.
+
+After the worfklow finishes execution, there will be two archives in the
+"Artifacts" section: one is the model that was trained and the other one is the
+SLSA provenance attached to the model.
+
+To verify the provenance, download both archives, unzip each and then run
+`slsa-verifier`, making sure to replace the `--source-uri` argument with the
+_path to your fork_. For example, for a PyTorch model:
+
+```console
+[...]$ slsa-verifier verify-artifact \
+       --provenance-path pytorch_model.pth.intoto.jsonl \
+       --source-uri github.com/google/model-transparency \
+       pytorch_model.pth
+Verified signature against tlog entry index 45172090 at URL: https://rekor.sigstore.dev/api/v1/log/entries/24296fb24b8ad77aba51e418ce36828f790c58a0f1304246a31eaadc35f36c2a0d03aabeb4b9ab07
+Verified build using builder "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@refs/tags/v1.9.0" at commit 831b9521692f564156a61998b2478378e7dc6f49
+Verifying artifact pytorch_model.pth: PASSED
+
+PASSED: Verified SLSA provenance
+```
+
+TODO: link to a PyTorch model provenance from running GHA within the repo, fix
+the output to match
+
```

Comment:
Something we might say here is that this example works on other cloud platforms like GCB, GitLab, etc.. or that there's WIP to make it as easy in other platforms too.

Reply:
Good idea. Added a note at the end, thank you